client/core/core.go

```diff
@@ -1680,24 +1710,37 @@ func (c *Core) SetWalletPassword(appPW []byte, assetID uint32, newPW []byte) err
 		wallet.Lock()
 	}
 
-	// Encrypt the password.
-	var encPW []byte
+	return nil
+}
+
+// setWalletPassword updates the (encrypted) password for the wallet.
+func (c *Core) setWalletPassword(wallet *xcWallet, newPW, encNewPW []byte) error {
+	newPasswordSet := len(newPW) > 0
+
+	// Check that the new password works. If the new password is empty, skip
+	// this step, since an empty password signifies an unencrypted wallet.
+	// TODO: find a way to verify that the wallet actually is unencrypted or
+	// otherwise does not require a password. Perhaps an
+	// asset.Wallet.RequiresPassword wallet method?
```

Comment:
I think we can work around this but I'll have to give you a suggested change in the a.m.  Just about out of steam for the day...

Reply:
Sounds great

Reply:
There's some interplay here with the existing unlock bug fixed in https://github.com/decred/dcrdex/pull/910.  I'm afraid it's going to be easier to get that in first.

I recognize the grief that causes you so I'm more than happy to rebase this (and force push to your branch if that's OK) and make a suggested edit after that.  But part of the issue that is preventing reconfigure from an encrypted->unencrypted wallet is that with the added `newWalletPW` arg, we no longer want to create the `db.Wallet` in `ReconfigureWallet` with `EncryptedPW: oldWallet.encPW` unless `newWalletPW == nil`.

Reply:
I see, makes total sense.

Please feel free to rebaae this PR and I can then add the suggested solution.

Reply:
Just did the rebase and will make some suggestions shortly. I suggest to git fetch and then reset hard your local branch to the new remote that was force pushed.  I did however add a few comments where I though changes were needed to resolve the issue.  Will follow up right away.

Reply:
@chappjc 
Sweet, got the updated branch locally 👍